### PR TITLE
[Mob 670] explore and add ui tests

### PR DESCRIPTION
**Pull request by Ayyanchira**
Added test method for inbox activity check:
This will serve as base/reference for new test methods which will be added to the project.

### CODE REVIEW

**Comment by vbabenkoru** [comment]
@Ayyanchira please change the target to `inbox`.

**Comment by Ayyanchira** [comment]
Thanks again for bringing the base branch into notice!

File: app/src/test/java/com/iterable/androidsdk/MainActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.junit.Assert.*;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
```
Comment (vbabenkoru) [inline app/src/test/java/com/iterable/androidsdk/MainActivityTest.java]: Doesn't look like this test is doing anything useful; do we need it?

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -21,7 +21,18 @@
 
     </android.support.design.widget.AppBarLayout>
 
-    <include layout="@layout/content_main" />
+    <include
+        android:id="@+id/include"
+        layout="@layout/content_main" />
+
+    <Button
```
Comment (vbabenkoru) [inline app/src/main/res/layout/activity_main.xml]: Do we need this change?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -18,6 +18,10 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name="com.iterable.iterableapi.ui.inbox.InboxActivity"
```
Comment (vbabenkoru) [inline app/src/main/AndroidManifest.xml]: This should be automatically added by the SDK during manifest merge, no need to add it explicitly into the Manifest.

File: app/src/main/res/inapp_payload_inbox_multiple.json [hunk 4]
```diff
@@ -0,0 +1,97 @@
+{
```
Comment (vbabenkoru) [inline app/src/main/res/inapp_payload_inbox_multiple.json]: This file is coped in multiple source folders, can you try to remove it in the ones where it's not being used?

File: app/src/androidTest/java/com/iterable/iterableapi/testapp/PathBasedQueueDispatcher.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
+        String path = request.getPath();
+        PathBasedResponse selectedResponse = null;
+        synchronized (responseQueue) {
+            for (PathBasedResponse response : responseQueue) {
+                if (path.startsWith(response.pathPrefix)) {
+                    selectedResponse = response;
+                }
```
Comment (vbabenkoru) [inline app/src/androidTest/java/com/iterable/iterableapi/testapp/PathBasedQueueDispatcher.java]: Add `break;` here, similar to the change in `iterableapi` module.

File: app/src/androidTest/java/com/iterable/iterableapi/MainActivityTest.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    @Rule
+    public ActivityTestRule rule = new ActivityTestRule<>(InboxActivity.class, false, false);
+
+    @Before
+    public void setUp() throws Exception {
+        server = new MockWebServer();
+        dispatcher = new PathBasedQueueDispatcher();
+        server.setDispatcher(dispatcher);
+        inAppHandler = mock(IterableInAppHandler.class);
+        customActionHandler = mock(IterableCustomActionHandler.class);
```
Comment (vbabenkoru) [inline app/src/androidTest/java/com/iterable/iterableapi/MainActivityTest.java]: Looks like some of these mocks are not used.

**Comment by Ayyanchira** [comment]
Force pushed with all necessary code changes.

**Pull request merged**